### DISCUSSION TITLE
authViaEmail : Authentication flow

### DISCUSSION THREAD

**Discussion by yepes** [pocketbase/pocketbase#500] [Q&A] [answered]
Hi, 

Does anybody have or want's what will be the best approach to keep users logged in when they log in via authViaEmail?

Reading the docs, I guess:

1) authViaEmail return an access token
2) store the access token in local storage
3) on the app first load, retrieve the access token if exists and try to refresh it.

Is that the correct approach? I don't like the idea of storing the token in local storage.

**Comment by ganigeorgiev** [comment, marked as answer]
Yeah, basically that's the expected flow. If you are using the [Javascript SDK](https://github.com/pocketbase/js-sdk) 1) and 2) are handled for you. For refresh, you can just call `client.users.refresh()`.

> I don't like the idea of storing the token in local storage.

Why? There is nothing wrong with storing the token in LocalStorage. The only problem with LocalStorage is if your app is XSS vulnerable. **But that's a big IF.** We've taken several measures to mitigate this by having a content security policies for both the `/api/file` endpoint and inside the Admin UI.

No matter whether you use LocalStorage, cookies, etc. it is highly recommended to add a CSP for your app. See the recommendations in https://github.com/pocketbase/js-sdk#security.

**Reply by YoRHa-A5** [reply]
So pb is using localstorage access tokens without refresh tokens stored in httponly cookies?

**Reply by ganigeorgiev** [reply]
@BunnyTheLifeguard  The token will be stored in LocalStorage if you use the JS SDK with the default AuthStore.

**Reply by YoRHa-A5** [reply]
@ganigeorgiev How is it with flutter? A couple of months ago I tried building an auth workflow with access- and refreshtokens separated in localstorage and httponly cookies but could not manage to get it to work since dart was not able to work with httponly cookies.

**Reply by ganigeorgiev** [reply]
@BunnyTheLifeguard 

The Dart SDK doesn't come with persistence store because it is too user-land specific. You can create a custom store that extends the default one and use any persistent mechanism you want (encrypted sharedpreferences, key-value store in a file/json, etc.). It's up to the developers.

You could also check this issue - https://github.com/pocketbase/dart-sdk/issues/13

**Reply by VladimirCores** [reply]
And what will be the key for 'token'/'user', and how could I check is user authenticated?

**Reply by VladimirCores** [reply]
Ok, looks like I found what was looking for ;) https://github.com/pocketbase/js-sdk#authstore

**Comment by yepes** [comment]
Ok thanks, so i guess the response of the authViaEmail should go to the localStorage, then retrieve everything when the page loads y call refresh. Thanks!

**Reply by yepes** [reply]
Actually, authStore is already populated.
